pkgs/misc/drivers/argononed/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, stdenv, fetchFromGitLab, dtc, installShellFiles }:
+
+stdenv.mkDerivation rec {
+  pname = "argononed";
+  version = "0.3.3";
+
+  src = fetchFromGitLab {
+    owner = "DarkElvenAngel";
+    repo = pname;
+    rev = "d50040115beece36a0259cef2248eff467286207";
```

Comment:
https://github.com/NixOS/nixpkgs/pull/166988#issuecomment-1086760575

Does not seem to match (probably because this is the 0.3.3 commit 🤔 ).

Reply:
Well, it's the latest commit before the bump to 0.3.4, so I think we can say it is 0.3.3?

Reply:
Fair enough.

I think this needs a comment tough. This is highly confusing.

Reply:
Just added a comment to the PR clarifying this. Should i comment the code itself as well?

Reply:
Yeah please. You're not the only one that will touch this package in the future.

Maybe an even better idea would be to set this to `unstable-YYYY-MM-DD` format. That is, unless we are sure this is the commit that points to 0.3.3 release from upstream (e.g. matches the tarball releases if they have some).

Reply:
It seems upstream maintains two branches: `0.3.x` (stable) and `0.4.x` (does not have all features yet). I think I'll go ahead and package `0.3.x-YYYY-MM-DD`, does that make sense?

Reply:
I am kinda against for custom version formats (we already have a lot of them).

What the main branch points too? If it is the `0.3.x` version I would say let's use the `unstable-YYYY-MM-DD` format and maybe add a comment explaining the situation.